ampd/src/commands/mod.rs

```diff
@@ -1,13 +1,129 @@
+use std::path::Path;
+
+use axelar_wasm_std::FnExt;
 use clap::Subcommand;
+use cosmos_sdk_proto::cosmos::auth::v1beta1::BaseAccount;
+use cosmos_sdk_proto::cosmos::base::abci::v1beta1::TxResponse;
+use cosmos_sdk_proto::cosmos::{
+    auth::v1beta1::query_client::QueryClient, tx::v1beta1::service_client::ServiceClient,
+};
+use cosmos_sdk_proto::Any;
+use cosmrs::AccountId;
+use error_stack::Result;
+use error_stack::ResultExt;
+use serde::{Deserialize, Serialize};
 use valuable::Valuable;
 
+use crate::broadcaster::{accounts::account, Broadcaster};
+use crate::config::Config as AmpdConfig;
+use crate::state::{self, StateUpdater};
+use crate::tofnd::grpc::{MultisigClient, SharableEcdsaClient};
+use crate::types::{PublicKey, TMAddress};
+use crate::{broadcaster, Error};
+use crate::{tofnd, PREFIX};
+
+pub mod bond_worker;
 pub mod daemon;
 pub mod worker_address;
 
 #[derive(Debug, Subcommand, Valuable)]
 pub enum SubCommand {
     /// Run the ampd daemon process (default)
     Daemon,
+    /// Bond the worker to the service registry contract
+    BondWorker(bond_worker::Args),
     /// Query the worker address
     WorkerAddress,
 }
+
+#[derive(Debug, Deserialize, Serialize, PartialEq)]
+pub struct Config {
+    pub cosmwasm_contract: TMAddress,
+    pub service_name: String,
+}
```

Comment:
seeing the config like this I have no idea what it's supposed to represent 😅 
should cosmwasm_contract be `service_registry_contract`?

Reply:
Also, it seems like service_name should be an argument instead? A worker can join multiple services, right?